test/typescript/dispatch.ts

```diff
@@ -1,12 +1,16 @@
 import {Dispatch, Action} from "../../index.d.ts";
 
 
-declare const dispatch: Dispatch;
-
+declare const dispatch: Dispatch<any>;
 
 const dispatchResult: Action = dispatch({type: 'TYPE'});
 
-
-type Thunk<O> = () => O;
+// thunk
+declare module "../../index.d.ts" {
+    export interface Dispatch<S> {
+        <R>(asyncAction: (dispatch: Dispatch<S>, getState: () => S) => R): R;
+    }
+}
```

Comment:
You're absolutely right, that  was my intent to define additional dispatch signature through module augmentation for `"redux"` module. And yes, it will work perfectly with node module resolution. I cannot write a test that uses `"redux"`, because within the package we haven't such module.